content/posts/2019-10-05-nestjs-prisma/index.mdx

```diff
@@ -0,0 +1,249 @@
+
+If at any point you get lost in this tutorial, please refer to [this tutorial's code on GitHub](https://github.com/AryanJ-NYC/nestjs-graphql-tutorial/tree/with-prisma). It should contain everything you need to catch up and integrate Prisma with NestJS.
+
+If you want to see this code in action, please visit https://enigmatic-journey-72826.herokuapp.com/graphql.
+
+Alright, let's crush some code.
+
+## Where We Left Off
+If you followed [my last tutorial](https://www.thewebdevcoach.com/stand-up-a-node-js-server-with-nestjs-including-typescript-and-graphql) correctly, you should end up with code that looks like [this GitHub branch](https://github.com/AryanJ-NYC/nestjs-graphql-tutorial/tree/nestjs-graphql). When you run `yarn start:dev`, a GraphQL server should spin up at [`http://localhost:3000/graphql`](http://localhost:3000) and you can make a `messages` query and a `createMessage` mutation. The server will look like [this one](https://enigmatic-journey-72826.herokuapp.com/graphql). If any of that is untrue, please refer to [the section above](#before-we-get-started).
+
+## Let's Get Ready for Prisma
+The first thing we need to do is instantiate a Prisma demo server and upload our data model to Prisma. For this, we'll use the [prisma-cli](https://github.com/prisma/prisma/tree/master/cli).
+
+```bash
+yarn global add prisma
+# or
+npm install -g prisma
+```
+
+With that installed, create a new prisma service. From your project folder, run the following command:
+
+```bash
+prisma init
+```
+
+You'll be given a choice of Prisma servers:
+![Photo of database choices given after prisma init is run](./img/prisma-server-choice.png)
+
+For this tutorial, we will set up a Demo server + MySQL database.
+
+The CLI will continue to ask configuration questions. Please be sure to use the `Prisma TypeScript Client` when asked:
+![Photo of client choices Prisma CLI provides](./img/prisma-client-choice.png)
+
+## Defining Our Data Model and Deploying
+After Prisma instantiation succeeds, two new files will be created at the root of our project:
+* `prisma.yml` - a configuration file Prisma uses to generate their schema, define schema output paths, etc.
+* `datamodel.prisma` - our project's data model which Prisma will use to generate schemas
+
+`datamodel.prisma` is initialized with a basic `User` model. We will replace it with the `Message` model (be sure to add the new `@id` directive that Prisma supplies us with) from our `src/schema.graphql` file:
+
+```graphql
+# datamodel.prisma
+
+type Message {
+  id: Int! @id
+  description: String!
+}
+```
+
+If everything worked as planned, Prisma will have generated a schema and TypeScript types definitions at `/src/generated/prisma-client`. Feel free to open up that folder and take a look around. Oddly enough, we will not be using this generated schema. We will use `prisma-binding` instead to generate a schema we can use with NestJS.
```

Comment:
This paragraph will change now that we won't generate any bindings with the CLI.